iothub/device/tests/IotHubDeviceClientTests.cs

```diff
@@ -1141,15 +1182,15 @@ public async Task IotHubDeviceClient_SendTelemetryAsync_Cancelled_ThrowsOperatio
             //arrange
             using var deviceClient = new IotHubDeviceClient(FakeConnectionString);
 
-            var mainProtocolHandler = Substitute.For<IDelegatingHandler>();
+            var mainProtocolHandler = new Mock<IDelegatingHandler>();
 
             // We will setup the main handler which can be either MQTT or AMQP or HTTP handler to throw
             // a cancellation token expiry exception (OperationCancelledException) to ensure that we mimic when a token expires.
             mainProtocolHandler
-                .When(x => x.SendTelemetryAsync(Arg.Any<TelemetryMessage>(), Arg.Any<CancellationToken>()))
-                .Do(x => { throw new OperationCanceledException(); });
+                .Setup(x => x.SendTelemetryAsync(It.IsAny<TelemetryMessage>(), It.IsAny<CancellationToken>()))
```

Comment:
Not super confident that this change is correct

Reply:
Why `Invoking` and not `Returns`? What's the difference?

Reply:
https://nsubstitute.github.io/help/callbacks/

Reply:
I wonder if this works the same if you use `Returns` instead.

`SendTelemetryAsync` returns something (a Task), but more importantly, we're just throwing an exception here. Seems that the callback/no-return-value-method doesn't really apply here.